marahgh/assignment3/wordSearch.cpp

```diff
@@ -0,0 +1,212 @@
+#include <iostream>

```

Comment:
It would make code more re-usable, if we split it into a `WordSearch.h` and `WordSearch.cpp` pair of files.
In fact, you might consider making the grid and dictionary their own pair of files.
This means you can reuse these more easily.
Also, at the moment, most of your methods are defined inline within the class declaration: this makes it harder to understand the structure of the class.

For example:

`grid.h`
```c++
/*The grid class, contains lower-case letters from whichever alphabet the dictionary uses.*/
class Grid {
  char** matrix;
  int rows;
  int cols;
  bool empty;

 public:
  /*creates an empty grid,
  a grid with the given dimensions filled by ' ' charachters*/
  Grid(int rows, int cols);
  /*receives a vector of the charachters to to fill the grid with,
  grid's rows number and grid's columns number.
  creates the grid if the dimensions are OK , otherwise creats an empty grid*/
  Grid(vector<char> chars, int rows, int cols);
  /*Fills the grid with the given values*/
  bool fillGrid(vector<char> chars);
  /*Grid's destructor*/
  ~Grid();
  /*returns the grid's rows number*/
  int getRows() { return rows; }
  /*returns the grid's columns number*/
  int getCols() { return cols; }
  /*return true if the grid is empty, and false otherwise*/
  bool isEmpty() { return empty; }
  char* operator[](int i);
};
```

`grid.cpp`
```c++
#include "grid.h"

Grid::Grid(int rows, int cols)
    : matrix(nullptr), rows(rows), cols(cols), empty(true) {
  matrix = new char*[rows];
  for (int i = 0; i < rows; i++) {
    matrix[i] = new char[cols];
  }
  vector<char> chars = vector<char>((rows * cols), ' ');
  empty = true;
  fillGrid(chars);
}

...
```

Looking at this, it also seems to me that `fillChars()` should not be public (which was not obvious in the previous form).